### DISCUSSION TITLE
Langium project generator including TypeScript LSP plugin.

### DISCUSSION THREAD

**Discussion by trusktr** [eclipse-langium/langium#981] [Ideas]
It would be sweet if the generator could include a mode where it not only bootstraps a standalone LSP extension for VS Code, but also a TypeScript LSP plugin. This would be useful for adding syntax and type checking overlaid on TypeScript. For example, someone may want to add additional syntax checking inside of ``` html`<div />` ``` template strings, etc, and would want to connect it with TypeScript awareness.

Another idea is adding additional type checks to TypeScript itself. For example `int` or `float` types that are aliased to `number`, but the TS LSP plugin can provide additional checks that validate if an int is being assigned to an int parameter, etc.

I'm going to try making a TS LSP plugin myself sometime soon, can can help with this if someone else hasn't gotten to it yet.

**Comment by msujew** [comment]
We're currently in the process of refactoring our yeoman generator, see https://github.com/langium/langium/pull/954. I believe after that, the package would be in good shape to also support such a setup 👍
